### ISSUE TITLE
Cilium KVStore failure status after a few minutes

### ISSUE THREAD

**Issue by myugan** [cilium/cilium#12904]
**General Information**

- Cilium version: `1.8.2 580d32ed8 2020-07-26T11:16:07+02:00 go version go1.14.3 linux/amd6`
- Kernel version: `Linux hostname 4.15.0-112-generic #113-Ubuntu SMP Thu Jul 9 23:41:39 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux`
- Orchestration system version: -
- Link to relevant artifacts: -

**How to reproduce the issue**

1. I use `etcd` with version `v3.4.10` also cilium on the host not running as container or kubernetes and this is my etcd systemd file:
```
[Unit]
Description=etcd key-value store
Documentation=https://github.com/etcd-io/etcd
After=network.target

[Service]
User=etcd
Type=notify
Environment=ETCD_DATA_DIR=/var/lib/etcd
Environment=ETCD_NAME=%m
ExecStart=/usr/local/bin/etcd \
  --listen-client-urls http://x.x.x.x:2379 \
  --advertise-client-urls http://x.x.x.x::2379 \
  --listen-peer-urls http://x.x.x.x::2380 \
  --initial-advertise-peer-urls http://x.x.x.x::2380 \
  --initial-cluster ${ETCD_NAME}=http://x.x.x.x::2380 \
  --initial-cluster-token xxxxx \
  --initial-cluster-state new \
  --log-level info \
  --logger zap \
  --log-outputs stderr
Restart=always
RestartSec=10s
LimitNOFILE=40000

[Install]
WantedBy=multi-user.target
```
Cilium systemd file:
```
[Unit]
Description=cilium
Documentation=http://docs.cilium.io
Requires=docker.service cilium-docker.service

[Service]
Type=simple
LimitCORE=infinity
ExecStart=/usr/bin/cilium-agent --enable-ipv6=false --kvstore etcd --kvstore-opt etcd.address=x.x.x.x:2379 -t vxlan
Restart=on-failure

[Install]
WantedBy=multi-user.target
```

2. After run etcd service and configured cilium to use etcd as kvstore it turn to success status when type `cilium status` like this:
```
KVStore:                Ok         etcd: 1/1 connected, lease-ID=7fdf73fbc0e37067, lock lease-ID=7fdf73fbc0e37069, has-quorum=true: x.x.x.x:2379 - 3.4.10 (Leader)
Kubernetes:             Disabled
Cilium:                 Ok         OK
NodeMonitor:            Disabled
Cilium health daemon:   Ok
IPAM:                   IPv4: 2/65535 allocated from 10.82.0.0/16,
Masquerading:           IPTables
Controller Status:      18/18 healthy
Proxy Status:           OK, ip 10.82.224.12, 0 redirects active on ports 10000-20000
Hubble:                 Disabled
Cluster health:         0/1 reachable   (2020-08-17T17:45:31+08:00)
  Name                  IP              Node        Endpoints      
  hostname (localhost)      x.x.x.x     reachable   unreachable
```
3. Wait a few minutes and check cilium status again, status **Ok** turn into **Failure**.
```
KVStore:                Failure    Err: quorum check failed 9 times in a row: 4m51.157975074s since last heartbeat update has been received
Kubernetes:             Disabled
Cilium:                 Failure    Kvstore service is not ready
NodeMonitor:            Disabled
Cilium health daemon:   Ok
IPAM:                   IPv4: 2/65535 allocated from 10.82.0.0/16,
Masquerading:           IPTables
Controller Status:      18/18 healthy
Proxy Status:           OK, ip 10.82.224.12, 0 redirects active on ports 10000-20000
Hubble:                 Disabled
```

**Comment by aanm** [comment]
@myugan do you have `cilium-operator` running in your cluster?

**Comment by myugan** [comment]
@aanm no, i install cilium from source code but there is no `cilium-operator` binary compiled,

**Comment by aanm** [comment]
@myugan you need to run `cilium-operator` as well.

```
$ make -C operator
$ ls -lah ./operator/cilium-operator
[...] ./operator/cilium-operator
```

**Comment by myugan** [comment]
Ok, i just copied that to `/usr/local/bin` but is it mandatory to use Kubernetes apiserver to run operator? because i run cilium on the host, or is it possible to pass argument to check apiserver?

**Comment by aanm** [comment]
@myugan yes. At the moment it is only possible to run Cilium and Cilium Operator in a Kubernetes environment. There was a PR that was specially made for 1.7 branch that allows Cilium to run without being dependent on the Cilium Operator. This functionality is not implemented in 1.8 nor in master. https://github.com/cilium/cilium/pull/12755

**Comment by myugan** [comment]
Okay thankyou for your information @aanm.

**Comment by myugan** [comment]
Solved by downgrade a version to `v1.7` I didn't get any error KVStore status after a few minutes.

**Comment by joestringer** [comment]
Looks like https://github.com/cilium/cilium/issues/12794 is tracking items for re-enabling this on later releases.
